pkg/build/controller/build_controller.go

```diff
@@ -130,6 +132,16 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 		return err
 	}
 
+	// Handle processing ValueFrom in build strategy environment variables
+	originalEnvVars := buildutil.GetEnvFromStrategy(build)
+	processedEnvVars, err := common.ProcessValueFrom(build, bc.KubeClient)
+	if err != nil {
+		build.Status.Phase = buildapi.BuildPhaseFailed
+		build.Status.Reason = buildapi.StatusReasonInvalidEnvironmentVariable
+		return err
+	}
+	*originalEnvVars = *processedEnvVars
```

Comment:
(Also, I remember the discussion but not the outcome of it: was the decision definitely to update the build object with the resolved env vars as well as the resulting pod, or was it just the latter?)